### PR TITLE
Return letters count summary report in csv format

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-626

### Change description ###
Updated reports controller to return the letters count summary in csv format.
Deleted the response model classes created for JSON response, 
as they are not used anymore.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/controllers/ReportsController.java [hunk 1]
```diff
@@ -28,24 +31,18 @@ public ReportsController(ReportsService reportsService) {
         this.reportsService = reportsService;
     }
 
-    @GetMapping(path = "/count-summary", produces = MediaType.APPLICATION_JSON_VALUE)
+    @GetMapping(path = "/count-summary", produces = MediaType.APPLICATION_OCTET_STREAM_VALUE)
     @ApiOperation("Retrieves uploaded letters count summary report")
-    public LettersCountSummaryResponse getCountSummary(
+    public ResponseEntity getCountSummary(
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/controllers/ReportsController.java]: is it `ResponseEntity<byte[]>` by any chance?

File: src/main/java/uk/gov/hmcts/reform/sendletter/controllers/ReportsController.java [hunk 2]
```diff
@@ -28,24 +31,18 @@ public ReportsController(ReportsService reportsService) {
...
         @RequestParam(name = "date") @DateTimeFormat(iso = DATE) LocalDate date
-    ) {
+    ) throws IOException {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/controllers/ReportsController.java]: Should this exception be handled independently and then custom one included in controller advice?

File: src/main/java/uk/gov/hmcts/reform/sendletter/util/CsvWriter.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import uk.gov.hmcts.reform.sendletter.model.out.LettersCountSummary;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class CsvWriter {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/util/CsvWriter.java]: missing `final` keyword ;)

File: src/main/java/uk/gov/hmcts/reform/sendletter/controllers/ReportsController.java (hunk 1 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/sendletter/controllers/ReportsController.java]: yes, it is. I'll update

**Pull request merged**